depth_image_proc/include/depth_image_proc/visibility.hpp

```diff
@@ -29,8 +29,8 @@
 // LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
 // ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
-#ifndef DEPTH_IMAGE_PROC__VISIBILITY_H_
```

Comment:
There is actually nothing C++ specific about this header, so let's just leave it as a `.h` file.

Reply:
@clalancette i thought with ROS2 we were [supposed to migrate to use `.hpp`](https://industrial-training-master.readthedocs.io/en/melodic/_source/session7/ROS1-to-ROS2-porting.html) (number 7 in that link)

Reply:
We aren't totally consistent on this, but in my view it should only be `.hpp` if it is a C++-only header file.  Leaving it as `.h` where it works in C means you can use this header file for C files as well (which is admittedly unlikely).

Reply:
so is this just your view or is it the ROS2 standard? Looking at the documentation it seems that the standard is to migrate to `.hpp`.  @peci1 makes a good point though that this change might break existing code, so what i've seen before is a blank `.h` file that includes the `.hpp` file, so "new" code can use the new `.hpp` standard, while old code wont be affected.

Reply:
ROS2 standard is `.hpp`. Since this is only going to the Galactic branch, I think changing them is fine as very few users will be affected.

Reply:
> so is this just your view or is it the ROS2 standard? Looking at the documentation it seems that the standard is to migrate to `.hpp`. @peci1 makes a good point though that this change might break existing code, so what i've seen before is a blank `.h` file that includes the `.hpp` file, so "new" code can use the new `.hpp` standard, while old code wont be affected.

Yeah, and we do that elsewhere we we ported ROS 1 C++ code that was in a `.h` file, and now we want to move it to `.hpp` to have it conform to ROS 2 standards.  In this case, though, there is really no need for the churn; the existing `.h` file will work for both C and C++.

> ROS2 standard is `.hpp`.

"ROS 2 standard is `.hpp` *for C++ code*".  Since this isn't purely C++ code (it is really C code that C++ understands), there's no reason in my view to change it.

That all said, this is a bit of bikeshedding and doesn't matter a huge amount at the end of the day.  I think it should be left as a `.h` file, but if the maintainers disagree we can change it.

Reply:
While I *mostly* agree with you, the kicker here is `extern "C"`. This only has meaning (and will ony compile) with a C++ compiler, not a C compiler. If we want to remove that and make the file a "pure C" header, then it should be included from other files as:

```c++
extern "C" {
#include <visibility.h>
}
```

Otherwise, as it is, it's a C++ header.

Reply:
```
#ifdef __cplusplus
extern "C"
{
#endif
```

Makes it work for both C and C++.

Reply:
You have a point.